### PR TITLE
Added instructions to the documentation for forking the main repo.

### PR DESCRIPTION
**Pull request by jacobslusser**
Added instructions to the "build from source" documentation for forking the main `dotnet/aspnet` repo on GitHub. This way, engineers are set up to start creating PRs if they want to.

Addresses #28192

CC: @captainsafia

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=33098) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=33098) before we can accept your contribution.<br/><br/>:x: jacobslusser [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=33098)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=33098) it.</sub>

**Comment by Pilchie** [comment]
Maybe a link to https://guides.github.com/activities/forking/ would be useful here?

**Comment by jacobslusser** [comment]
In lieu of the new instructions or as supplement to them?

**Comment by Pilchie** [comment]
I was thinking as a supplement.

File: docs/BuildFromSource.md [hunk 1]
```diff
@@ -9,27 +9,60 @@ For more info on issues related to build infrastructure and ongoing work, see <h
...
+
+You can verify the `upstream` remote has been set correctly.
+
+```powershell
+git remote -v
+> origin    https://github.com/YOUR_USERNAME/aspnetcore (fetch)
+> origin    https://github.com/YOUR_USERNAME/aspnetcore (push)
+> upstream  https://github.com/dotnet/aspnetcore.git (fetch)
+> upstream  https://github.com/dotnet/aspnetcore.git (push)
+```
```
Comment (captainsafia) [inline docs/BuildFromSource.md:53]: We can also add a note about GitHub's new fork syncing feature here.

https://github.blog/changelog/2021-05-06-sync-an-out-of-date-branch-of-a-fork-from-the-web/

**Review by captainsafia — COMMENTED** [review]
Looks good overall. Added one other note about adding a reference to GitHub's new feature for keeping forks in sync as a non-CLI alternative.

**Review by captainsafia — APPROVED** [review]

**Pull request merged**